keyserver/storage/postgres/key_changes_table.go

```diff
@@ -55,7 +57,24 @@ func NewPostgresKeyChangesTable(db *sql.DB) (tables.KeyChanges, error) {
 		db: db,
 	}
 	_, err := db.Exec(keyChangesSchema)
-	return s, err
+	if err != nil {
+		return s, err
+	}
+
+	// TODO: Remove when we are sure we are not having goose artifacts in the db
+	// This forces an error, which indicates the migration is already applied, since the
+	// column partition was removed from the table
+	var count int
+	err = db.QueryRow("SELECT partition FROM keyserver_key_changes LIMIT 1;").Scan(&count)
```

Comment:
Can we not just select from the goose_db_version table for the migration version?

Reply:
Checking `goose_db_version` will not work for users starting with Dendrite releases once this is merged or users manually removing the table., so I'd rather keep this check in. It's not ideal, for sure.

Reply:
If users have manually removed the goose db version table then uh, that's on them. For Dendrite users starting in the next release, we just need to make sure it's non-fatal that if the table doesn't exist then we know that they have been migrated already.